stretch_gazebo/nodes/keyboard_teleop_gazebo

```diff
@@ -0,0 +1,299 @@
+            self.step_size = 'medium'
+        if c == 's' or c == 'S':
+            rospy.loginfo('process_keyboard.py: changing to SMALL step size')
+            self.step_size = 'small'
+        if c == 'q' or c == 'Q':
+            rospy.loginfo('keyboard_teleop exiting...')
+            rospy.signal_shutdown('Received quit character (q), so exiting')
+
+        ####################################################
+
+        return command
+
+
+class KeyboardTeleopNode:
+
+    def __init__(self):
+
+        self.keys = GetKeyboardCommands()
+        self.rate = 10.0
+        self.joint_state = None
+        self.twist = Twist()
+
+    def joint_states_callback(self, joint_state):
+        self.joint_state = joint_state
+
+    def send_command(self, command):
+        joint_state = self.joint_state
+        self.trajectory_client_selector(command)
+        if (joint_state is not None) and (command is not None):
+            if 'translate_mobile_base' == command['joint'] or 'rotate_mobile_base' == command['joint']:
+                self.cmd_vel_pub.publish(self.twist)
+                return 0
+            point = JointTrajectoryPoint()
+            point.time_from_start = rospy.Duration(0.2)
+            trajectory_goal = FollowJointTrajectoryGoal()
+            trajectory_goal.goal_time_tolerance = rospy.Time(1.0)
+
+            joint_name = command['joint']
+
+            if joint_name in ['joint_lift', 'joint_wrist_yaw', 'joint_head_pan', 'joint_head_tilt']:
+                trajectory_goal.trajectory.joint_names = [joint_name]
+                joint_index = joint_state.name.index(joint_name)
+                joint_value = joint_state.position[joint_index]
+                delta = command['delta']
+                new_value = joint_value + delta
+                point.positions = [new_value]
+            elif joint_name in ["joint_gripper_finger_left", "wrist_extension"]:
+                if joint_name == "joint_gripper_finger_left":
+                    trajectory_goal.trajectory.joint_names = ['joint_gripper_finger_left', 'joint_gripper_finger_right']
+                else:
+                    trajectory_goal.trajectory.joint_names = ['joint_arm_l0','joint_arm_l1', 'joint_arm_l2', 'joint_arm_l3']
+                positions = []
+                for j_name in trajectory_goal.trajectory.joint_names:
+                    joint_index = joint_state.name.index(j_name)
+                    joint_value = joint_state.position[joint_index]
+                    delta = command['delta']
+                    new_value = joint_value + delta/len(trajectory_goal.trajectory.joint_names)
+                    positions.append(new_value)
+                point.positions = positions
+
+            trajectory_goal.trajectory.points = [point]
+            trajectory_goal.trajectory.header.stamp = rospy.Time.now()
+            self.trajectory_client.send_goal(trajectory_goal)
+            self.trajectory_client.wait_for_result()
```

Comment:
Did adding the `wait_for_result` resolve the undetected key presses issue described in this [post](https://forum.hello-robot.com/t/error-when-setting-up-gazebo-with-moveit-and-stretch/218/10)?

Reply:
The `wait_for_result` did not make any effect on the previous undetected key presses issue. But the key presses are detected but delayed due to a loop delay between the stretch controllers spawned from Gazebo and the keyboard teleop node.

Reply:
Okay, let's merge and we can come back to this issue if needed.